src/test/java/org/checkstyle/autofix/recipe/HeaderTest.java

```
@@ -42,15 +40,14 @@ protected Recipe getRecipe() throws CheckstyleException {
         final String configPath = "src/test/resources/org/checkstyle/autofix/recipe/header"
                 + "/config.xml";
 
-        final Configuration config = ConfigurationLoader.loadConfiguration(
-                configPath, new PropertiesExpander(new Properties())
-        );
+        final CheckConfiguration config = ConfigurationLoader.loadConfiguration(configPath);
 
         final List<CheckstyleViolation> violations =
                 CheckstyleReportParser.parse(Path.of(reportPath));
 
-        return new Header(violations,
-                extractCheckConfiguration(config, "Header"), getCharset(config));
+        final CheckConfiguration checkConfig = config.getChild("Header");
```

1. Could you rename `getChild` to `getChildConfig`
2. Using such hardcoded values `"Header"` may be error prone, any type or multiple places and we are in trouble. Let's handle this separately, as an option we can create an enum and put there all checks the we have recipes for. Please create and issue for this

---

We're already planning to shift to using `CheckstyleAutoFix` directly, so I don't think we'll end up using `CheckEnum`.
Adding it for now to make this mergable, but it's likely to be removed later.